dlt/destinations/filesystem/filesystem.py

```diff
@@ -0,0 +1,119 @@
+
+        root_path = Path(fs_path).joinpath(dataset_name)
+
+        if write_disposition == 'merge':
+            write_disposition = 'append' if has_merge_keys else 'replace'
+
+        if write_disposition == 'replace':
+            job_info = LoadStorage.parse_job_file_name(file_name)
+            glob_path = str(root_path.joinpath(f"{schema_name}.{job_info.table_name}.*"))
+            items = fs_client.glob(glob_path)
+            if items:
+                fs_client.rm(items)
+
+        destination_file_name = LoadFilesystemJob.make_destination_filename(file_name, schema_name, load_id)
+        fs_client.put_file(file_path, root_path.joinpath(destination_file_name))
+
+    @staticmethod
+    def make_destination_filename(file_name: str, schema_name: str, load_id: str) -> str:
+        job_info = LoadStorage.parse_job_file_name(file_name)
+        return f"{schema_name}.{job_info.table_name}.{load_id}.{job_info.file_id}.{job_info.file_format}"
+
+    def state(self) -> TLoadJobState:
+        return "completed"
+
+    def exception(self) -> str:
+        raise NotImplementedError()
+
+
+class FilesystemClient(JobClientBase):
+    """filesystem client storing jobs in memory"""
+
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+    fs_client: AbstractFileSystem
+    fs_path: str
+
+    def __init__(self, schema: Schema, config: FilesystemClientConfiguration) -> None:
+        super().__init__(schema, config)
+        self.fs_client, self.fs_path = client_from_config(config)
+        self.config: FilesystemClientConfiguration = config
+
+    @property
+    def dataset_path(self) -> Path:
+        return Path(self.fs_path).joinpath(self.config.dataset_name)
+
+    def initialize_storage(self, staging: bool = False, truncate_tables: Iterable[str] = None) -> None:
+        self.fs_client.makedirs(Path(self.fs_path).joinpath(self.config.dataset_name), exist_ok=True)
+
+    def is_storage_initialized(self, staging: bool = False) -> bool:
+        return self.fs_client.isdir(self.config.dataset_name)  # type: ignore[no-any-return]
+
+    def start_file_load(self, table: TTableSchema, file_path: str, load_id: str) -> LoadJob:
+        has_merge_keys = any(col['merge_key'] or col['primary_key'] for col in table['columns'].values())
+        return LoadFilesystemJob(
+            file_path=file_path,
+            write_disposition=table['write_disposition'],
+            has_merge_keys=has_merge_keys,
+            schema_name=self.schema.name,
+            dataset_name=self.config.dataset_name,
+            load_id=load_id,
+            fs_client=self.fs_client,
+            fs_path=self.fs_path
+        )
+
+    def restore_file_load(self, file_path: str) -> LoadJob:
+        return EmptyLoadJob.from_file_path(file_path, "completed")
+
+    def create_merge_job(self, table_chain: Sequence[TTableSchema]) -> NewLoadJob:
+        return None
+
+    def complete_load(self, load_id: str) -> None:
```

Comment:
I was looking at our use cases and actually it would be useful to create a file here. the reason: when the file is present we now that load to the bucket was finalized and we can use the loaded data. You could probably execute LoadFileSystemJob. write_disposition: append, the file can be empty, and the table name is LOADS_TABLE_NAME (_dlt_loads)

Reply:
https://github.com/dlt-hub/dlt/pull/342/commits/63a97ec6b0e19f25869f213fac6d9325a98e14d0#diff-524215919bce1cfd1cd9bae6d4ae82c9f753d4b3ffb1215a418dd06277113a5dR107-R110  
Will this do? Just creates empty file named `{schema_name}.{_dlt_loads}.{load_id}`  . Figure extension and file ID is not needed since it's just a marker per load ID?

Reply:
no, it is not needed. thx!